contributingGuides/TS_STYLE.md

```diff
@@ -0,0 +1,403 @@
+  ```
+
+<a name="convension-proptypes-and-defaultprops"></a><a name="1.11"></a>
+
+- [1.11](#convension-proptypes-and-defaultprops) **`propTypes` and `defaultProps`**: Do not use them. Use object destructing to assign default values if necessary.
+
+  > Refer to [the propTypes Migration Table](./PROPTYPES_CONVERSION_TABLE.md) on how to type props based on existing `propTypes`.
+
+  ```tsx
+  type GreetingProps = {
+    greeting: string;
+    name: string;
+  };
+
+  function Greeting({ greeting = "hello", name = "world" }: ComponentProps) {
+    <Text>{`${greeting}, ${name}`}</Text>;
+  }
+  ```
+
+<a name="convension-utility-types"></a><a name="1.12"></a>
+
+- [1.12](#convension-utility-types) **Utility Types**: Use types from [TypeScript utility types](https://www.typescriptlang.org/docs/handbook/utility-types.html) and [`type-fest`](https://github.com/sindresorhus/type-fest) when possible.
+
+  ```ts
+  type Foo = {
+    bar: string;
+  };
+
+  // good
+  type ReadOnlyFoo = Readonly<Foo>;
+
+  // bad
+  type ReadOnlyFoo = {
+    readonly [Property in keyof Foo]: Foo[Property];
+  };
+  ```
+
+<a name="convension-object-type"></a><a name="1.13"></a>
+
+- [1.13](#convension-object-type) **`object`**: Don't use `object` type.
+
+  > Why? `object` refers to "any non-primitive type," not "any object". Typing "any non-primitive value" is not commonly needed.
+
+  ```ts
+  // bad
+  const foo: object = [1, 2, 3]; // TypeScript does not error
+  ```
+
+  If you know that the type of data is an object but don't know what properties or values it has beforehand, use `Record<string, unknown>`.
+
+  > Even though `string` is specified as a key, `Record<string, unknown>` type can still accepts objects whose keys are numbers or symbols. This is because number and
+
+  ```ts
+  function logObject(object: Record<string, unknown>) {
+    for (const [key, value] of Object.entries(object)) {
+      console.log(`${key}: ${value}`);
+    }
+  }
+  ```
+
+<a name="convension-export-prop-types"></a><a name="1.14"></a>
+
+- [1.14](#convension-export-prop-types) **Prop Types**: Define and export prop types for components. Use exported prop types instead of grabbing the prop type from a component.
+
+  > Why? Exporting prop types aids reusability.
+
+  ```tsx
+  // MyComponent.tsx
+  export type MyComponentProps = {
+    foo: string;
+  };
+
+  export default function MyComponent({ foo }: MyComponentProps) {
+    return <Text>{foo}</Text>;
+  }
+
+  // bad
+  import { ComponentProps } from "React";
+  import MyComponent from "./MyComponent";
+  type MyComponentProps = ComponentProps<typeof MyComponent>;
+
+  // good
+  import MyComponent, { MyComponentProps } from "./MyComponent";
+  ```
+
+<a name="convension-file-organization"></a><a name="1.15"></a>
+
+- [1.15](#convension-file-organization) **File organization**: In modules with platform-specific implementations, create `types.ts` to define shared types. Import and use shared types in each platform specific files.
```

Comment:
What about reusable type definitions, should we add a subsection or bullet point for it?

> Reusable type definitions, such as models (e.g. Report), must have their own file and be placed in a shared directory, e.g. `src/types/`.

Reply:
added as its own section